### PR TITLE
add functionality to find node by id

### PR DESCRIPTION
**Pull request by SteffiPeTaffy**
recursively visits all nodes in all subgraphs to find node by its id.

### CODE REVIEW

File: graph.go [hunk 1]
```diff
@@ -265,3 +265,25 @@ func appendSortedMap(m map[string]interface{}, mustBracket bool, b io.Writer) {
...
+		subGraph.VisitNodes(callback)
+	}
+}
+
+// FindNodeById return node by id
+func (g Graph) FindNodeById(id string) (foundNode Node, found bool) {
+	g.VisitNodes(func(node Node) {
+		if node.id == id {
+			found = true
+			foundNode = node
```
Comment (emicklei) [inline graph.go]: you can put an early return here ; no need to visit all nodes after you have found the first one

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,3 @@
-module github.com/emicklei/dot
```
Comment (emicklei) [inline go.mod:1]: this change should not have been committed

Comment (SteffiPeTaffy) [reply go.mod:1]: sorry, was by mistake. I will undo this commit.

**Comment by SteffiPeTaffy** [comment]
Sorry, I accidentally chose my master branch for the PR. I'll close this an create a new PR from different branch.

**Pull request closed without merging**